mock/mock.go

```diff
@@ -209,6 +209,12 @@ func (m *Mock) On(methodName string, arguments ...interface{}) *Call {
 	return c
 }
 
+// IsCallExpected checks if a method call is expected.
+func (m *Mock) IsCallExpected(method string, arguments ...interface{}) bool {
+	found, _ := m.findExpectedCall(method, arguments...)
+	return found >= 0
+}
```

Comment:
It will basically have to duplicate the findExpectedCall() logic. But we don't have access to the m.mutex. And it will also break our code if there is any change to the findExpectedCall() in the future.